src/Classes/ImportTab.lua

```diff
@@ -170,48 +170,106 @@ You can get this from your web browser's cookies while logged into the Path of E
 	self.controls.generateCodeCopy.enabled = function()
 		return #self.controls.generateCodeOut.buf > 0
 	end
-	self.controls.generateCodePastebin = new("ButtonControl", {"LEFT",self.controls.generateCodeCopy,"RIGHT"}, 8, 0, 140, 20, "Share with Pastebin", function()
-		local id = LaunchSubScript([[
-			local code, proxyURL = ...
-			local curl = require("lcurl.safe")
-			local page = ""
-			local easy = curl.easy()
-			easy:setopt_url("https://pastebin.com/api/api_post.php")
-			easy:setopt(curl.OPT_POST, true)
-			easy:setopt(curl.OPT_POSTFIELDS, "api_dev_key=c4757f22e50e65e21c53892fd8e0a9ff&api_paste_private=1&api_option=paste&api_paste_code="..code)
-			easy:setopt(curl.OPT_ACCEPT_ENCODING, "")
-			if proxyURL then
-				easy:setopt(curl.OPT_PROXY, proxyURL)
-			end
-			easy:setopt_writefunction(function(data)
-				page = page..data
-				return true
-			end)
-			easy:perform()
-			easy:close()
-			if page:match("pastebin.com") then
-				return page
-			else
-				return nil, page
+
+	-- For new export website, add another entry to this table with data filled out
+	-- Label for dropdown, id for err, subScript for POST call, codeOut for SetText
+	-- codeOut if we need to append anything (*in front of) response, pastebin returns full link, pobbin returns only id for example
+	local exportWebsiteList = {
+		{
+			label = "Pastebin.com",
+			id = "Pastebin",
+			codeOut = "",
+			subScript = function()
+				return LaunchSubScript([[
+					local code, proxyURL = ...
+					local curl = require("lcurl.safe")
+					local page = ""
+					local easy = curl.easy()
+					easy:setopt_url("https://pastebin.com/api/api_post.php")
+					easy:setopt(curl.OPT_POST, true)
+					easy:setopt(curl.OPT_POSTFIELDS, "api_dev_key=c4757f22e50e65e21c53892fd8e0a9ff&api_paste_private=1&api_option=paste&api_paste_code="..code)
+					easy:setopt(curl.OPT_ACCEPT_ENCODING, "")
+					if proxyURL then
+						easy:setopt(curl.OPT_PROXY, proxyURL)
+					end
+					easy:setopt_writefunction(function(data)
+						page = page..data
+						return true
+					end)
+					easy:perform()
+					easy:close()
+					if page:match("pastebin.com") then
+						return page
+					else
+						return nil, page
+					end
+				]], "", "", self.controls.generateCodeOut.buf, launch.proxyURL)
+			end
+		},
+		{
+			label = "pobb.in",
+			id = "pobbin",
+			codeOut = "https://pobb.in/",
+			subScript = function()
+				return LaunchSubScript([[
+					local code, proxyURL = ...
+					local curl = require("lcurl.safe")
+					local page = ""
+					local easy = curl.easy()
+					easy:setopt_url("https://pobb.in/pob/")
+					easy:setopt(curl.OPT_POST, true)
+					easy:setopt(curl.OPT_POSTFIELDS, code)
```

Comment:
Could probably have lines 1 and 3 here be the only differences in the functions, using the 200 response for pastebin as well.  Then we could even use the same table `importWebsiteList` that is used for import so all the data for each site is in one place

Reply:
Ahh, completely agree, would you say putting this LaunchSubScript snippet back into the Share function (only once total overall) but having those two lines dynamically filled from the (import) table? Assuming it can work like that...

Reply:
I believe it can work like that, so yeah that sounds like a solid plan.